components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/utils/APIUtil.java

```diff
@@ -2676,6 +2677,72 @@ public static String replaceEmailDomain(String input) {
         return input;
     }
 
+    /**
+     * This method is used to get the APIProvider instance for a given username
+     *
+     * @param context API Context of the API
+     * @throws APIManagementException If the context is not valid
+     */
+    public static void validateAPIContext(String context) throws APIManagementException {
+        Pattern pattern = Pattern.compile(contextRegex);
+
+        if (context == null || context.isEmpty()) {
+            log.error("Invalid Context: Context cannot be empty or null");
+            throw new APIManagementException("Invalid Context: Context cannot be empty or null");
+        }
+
+        if (context.endsWith("/")) {
+            log.error("Invalid Context: Context cannot end with /");
+            throw new APIManagementException("Invalid Context: Context cannot end with /");
+        }
+
+        Matcher matcher = pattern.matcher(context);
+
+        // if the context has allowed characters
+        if (matcher.matches()) {
+            context = context.startsWith("/") ? context : "/".concat(context);
+            String split[] = context.split("/");
+
+            for (String param : split) {
+                if (!param.equals("{version}") && (param.contains("{version") || param.contains("{") || param.contains(
```

Comment:
Lets use org.wso2.carbon.apimgt.impl.APIConstants#VERSION_PLACEHOLDER instead of {version}